blog/models.py

```diff
@@ -3,25 +3,42 @@
 
 class Blog(models.Model):
     title = models.CharField(max_length=100, blank=True, default='')
-    description = models.TextField(max_length=1500)
+    description = models.TextField(max_length=50000)
     created = models.DateTimeField(auto_now_add=True)
     owner = models.ForeignKey('auth.User', related_name='blog', on_delete=models.CASCADE, null=True)
     image = models.FileField(blank=True, null=True)
 
+    @property
+    def votes(self):
+        upvotes = UserVote.objects.filter(blog=self, vote_type='up').count()
+        downvotes = UserVote.objects.filter(blog=self, vote_type='down').count()
+        return upvotes - downvotes
+
+    def upvote(self, user):
+        vote, created = self.post_votes.get_or_create(user=user, blog=self)
+        if not created and vote.vote_type == 'up':
+            return 'already_voted'
+
+        vote.vote_type = 'up'
+        vote.save()
+        return 'ok'
```

Comment:
`You have successfully up voted this blog post.`

Reply:
Done